### PR TITLE
LG-3569: dropdown borders

### PR DESCRIPTION
**Pull request by shaneeza**
## ✍️ Proposed changes

Updates `SearchInput` and `Select` dropdowns. 
Updates to `Select` will update `NumberInput`, `Pagination`, and `LanguageSwitcher`.
`Combobox` already has the correct border.

🎟 _Jira ticket:_ [LG-3569](https://jira.mongodb.org/browse/LG-3569)

### For bug fixes, new features & breaking changes

- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)
- [x] I have run `yarn changeset` and documented my changes

### For new components

- [ ] I have added my new package to the global tsconfig
- [ ] I have added my new package to the Table of Contents on the global README
- [ ] I have verified the Live Example looks as intended on the design website.

### CODE REVIEW

**Comment by bruugey** [comment]
Shocked that this caused no flag in Chromatic

File: packages/select/src/styleSets.ts [hunk 1]
```diff
@@ -70,7 +70,7 @@ export const colorSets: Record<Theme, ColorSet> = {
 
   [Theme.Dark]: {
     menu: {
-      border: palette.gray.base,
+      border: 'rgba(255, 255, 255, 0)',
```
Comment (bruugey) [inline packages/select/src/styleSets.ts:73]: palette.white?

File: packages/search-input/src/SearchResultsMenu/SearchResultsMenu.style.ts [hunk 2]
```diff
@@ -13,9 +13,11 @@ export const searchResultsMenuStyles = css`
 export const searchResultsMenuThemeStyles: Record<Theme, string> = {
   [Theme.Light]: css`
     background-color: ${palette.white};
+    border: 1px solid ${palette.gray.light2};
   `,
   [Theme.Dark]: css`
     background-color: ${palette.gray.dark3};
+    border: 1px solid rgba(255, 255, 255, 0);
```
Comment (bruugey) [inline packages/search-input/src/SearchResultsMenu/SearchResultsMenu.style.ts:20]: palette.white?

File: packages/select/src/styleSets.ts (hunk 1 above)
Comment (bruugey) [reply packages/select/src/styleSets.ts:73]: if not, i think this should at least be a css`` string

File: .changeset/gorgeous-queens-wave.md [hunk 3]
```diff
@@ -0,0 +1,6 @@
+---
+'@leafygreen-ui/search-input': patch
+'@leafygreen-ui/select': patch
+---
+
+Add a 1px `gray.light2` border to light mode dropdowns and a 1px `translucent` border to dark mode dropdowns to match Figma specs.
```
Comment (shaneeza) [inline .changeset/gorgeous-queens-wave.md:6]: transparent

File: packages/search-input/src/SearchResultsMenu/SearchResultsMenu.style.ts (hunk 2 above)
Comment (shaneeza) [reply packages/search-input/src/SearchResultsMenu/SearchResultsMenu.style.ts:20]: This is for darkMode so i left it transparent but i could make it mongo black

**Comment by shaneeza** [comment]
> Shocked that this caused no flag in Chromatic

yeah, it should show a 1px shift in the dropdowns

**Review by spark33 — APPROVED** [review]

**Pull request merged**